src/main/java/org/betonquest/betonquest/events/DropItemEvent.java

```diff
@@ -112,42 +121,48 @@ public void onPlayerJoin(final PlayerJoinEvent event) {
 
     @EventHandler(ignoreCancelled = true)
     public void onMergingItem(final ItemMergeEvent event) {
-        for (final Entity item : entityPlayerMap.keySet()) {
-            if (event.getEntity().equals(item)) {
-                event.setCancelled(true);
-            }
-        }
-        for (final Entity item : indestructibleItem) {
-            if (event.getEntity().equals(item)) {
-                event.setCancelled(true);
+        //TODO: Define default behavior: Shouldn't private items always be protected from being converted into public
+        //      items by being merged with another players private items?
+        if (isProtected) {
+            for (final Entity item : entityPlayerMap.keySet()) {
+                if (event.getEntity().equals(item)) {
+                    event.setCancelled(true);
+                }
             }
+
         }
     }
 
     @EventHandler(ignoreCancelled = true)
     public void onItemDespawn(final ItemDespawnEvent event) {
-        for (final Entity item : indestructibleItem) {
-            if (event.getEntity().equals(item) && event.getEntity().getType() == EntityType.DROPPED_ITEM) {
-                event.setCancelled(true);
+        if (isProtected) {
+            for (final Entity item : entityPlayerMap.keySet()) {
+                if (event.getEntity().equals(item) && event.getEntity().getType() == EntityType.DROPPED_ITEM) {
+                    event.setCancelled(true);
+                    //TODO: What happens to the items live time? It might have to be reset so this event doesn't fire
```

Comment:
IDK what this mean, isnt this event only get called if item getdespawn (5 minute)

Reply:
Item entities have an `age` attribute. 
![grafik](https://user-images.githubusercontent.com/40303883/135411099-a8fdab94-e012-4b37-a847-21f6840f96fd.png)
The question is about whether the age attribute get's reset to 0 when the event is cancelled. Otherwise the event may be fired each tick when the age is > 6000. This would be bad for performance.

Reply:
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Entity.html#setTicksLived%28int%29 will this work? set it to -32768 like that image said

Reply:
I don't know, the screenshots describes the vanilla but not the Bukkit/Spigot/Paper behaviour. You would have to test that.